### DISCUSSION TITLE
App Installation

### DISCUSSION THREAD

**Discussion by LDprg** [winapps-org/winapps#47] [Ideas]
We should start thinking about the installation of apps in the rewrite. Because the connect command will only start a program but doesn't create shurtcuts, etc.

However how do we detect apps?
Here some ideas:
- Like the legacy just shipping some definition files
- Fetching apps from a git repo that gets updated
- Detect apps directly from the vm

**Comment by oskardotglobal** [comment]
I feel like detecting apps directly from the vm would be our best bet here.
I don't actually know how windows does this, but I feel like all installed apps would be in a registry somewhere. We could even send the icon from windows directly to the linux machine for the desktop entries which would solve a lot of problems
We could still have "override files" which could add e.g. command line flags to the windows executable or add flags to freerdp 

We definitely need some sort of tunnel to the vm to be able to run commands etc.
I've tried setting up openssh for windows via unattended.xml without any success

**Reply by LDprg** [reply]
The problem with detecting the apps from windows is that there is a lot of bloat (feels more like than actually needed software). This would cause a long list with a lot of unneeded and redundant stuff. Another thing is that there is no reliable way to read out the installed apps from a windows desktop. Because apps are mostly installed via Installers and the Microsoft store it is somewhat chaotic and splitted where apps are registered or not. The main reason is that most software is installed via Installers, in addition there is also the Microsoft store and winget. 

So it is not straight forward and we should also consider portable apps.

**Reply by Shaarigan** [reply]
1. Apps are only put into registry if they were installed from an official installer that sets them. If it doesn't then they won't show up here and the Software App cannot uninstall them
2. There is a special directory for Desktop shortcuts for every and all users
3. There is a special directory for the Quick Launch bar at %APPDATA%\Microsoft\Internet Explorer\Quick Launch

You could write a shell extension that creates a popup menu entry for executables and provide a checkmark if that executable should be displayed in the Host's application menu. So it eliminates any trouble between desktop shortcuts, protable apps and store stuff

**Reply by oskardotglobal** [reply]
Ideally, the menu to manage which apps show up on the host would be .... well on the host, but I like the idea

**Reply by Shaarigan** [reply]
True but on the other hand I'd not want my app menu being bloated by all those irrelevant Windows apps like Phone, Mail or Calculator when I already have those on my host system. Waydroid does the same thing and just pushes every app installed onto the host system, they have way less Apps on their Lineage Android build but it also contains a calculator and some other stuff so I'd appreciate a more fine grained control over what will be put onto the host.

Similar to what Apps I pin to the task bar in Windows

**Reply by oskardotglobal** [reply]
What I'm talking about is having a menu as a part of winapps-gui where you can literally select the apps that are created as desktop integrations
which all of them disabled by default of course

**Reply by LDprg** [reply]
> What I'm talking about is having a menu as a part of winapps-gui where you can literally select the apps that are created as desktop integrations
> which all of them disabled by default of course

Yeah but how are we doing on cli?
I mean we could program a tui, but I would prefer a file based solution, where the user just edit these ones. The gui could just create those files via menue. This way we also keep compatibility from cli to gui.

This would also allow us to provide a default config (distribution could be done via git).

**Reply by LDprg** [reply]
For the gui we could make a auto detection, which would still end up creating those config files.

For the cli the user had to do it manually or just use the gui for this purpose.
